sound/soc/sof/intel/hda-codec.c

```diff
@@ -173,8 +175,10 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 		 * handle ret==0 (no driver bound) as an error, but pass
 		 * other return codes without modification
 		 */
-		if (ret == 0)
+		if (ret == 0) {
+			snd_hdac_ext_bus_device_exit(hdev);
 			ret = -ENOENT;
+		}
```

Comment:
This isn't something introduced by this patch, but this code means, that in case of success this function returns 1, which isn't very common for a `.probe()` style function. I see that this function is only called from one location and there its return code is checked for `< 0` but maybe it would be good to change this to return 0 on success.

Reply:
yes, I get your idea.